libpromises/evalfunction.c

```diff
@@ -2699,6 +2699,70 @@ bool EvalContextVariablePutSpecialEscaped(EvalContext *ctx, SpecialScope scope,
     return EvalContextVariablePutSpecial(ctx, scope, lval, value, type, tags);
 }
 
+/*********************************************************************/
+
+static JsonElement* ExecJSON_Pipe(const char *cmd, JsonElement *container)
+{
+    IOData io = cf_popen_full_duplex(cmd, false, false);
+
+    if (io.write_fd == -1 || io.read_fd == -1)
+    {
+        Log(LOG_LEVEL_INFO, "An error occurred while communicating with '%s'", cmd);
+        return NULL;
```

Comment:
Please look at `PipeWriteData()` which doesn't. This code is copied from there, so if this needs fixing, then `PipeWriteData()` needs fixing too.